probes/leveldown-probe.js

```diff
@@ -13,97 +13,129 @@
+    var lvldownObj = target.apply(null, arguments);
+    lvldownObj._ddProbeAttached_ = true;
+    aspectLvldownMethod(lvldownObj, methods, that);
+    return lvldownObj;
+  };
+  return newTarget;
+};
 
 /*
  * Lightweight metrics probe for leveldown queries
- * 
+ *
  * These provide:
  * 		time:		time event started
  * 		method: 	leveldown method being executed
- *		key:		The key being used for a call to `get`, `put` or `del` 
+ *		key:		The key being used for a call to `get`, `put` or `del`
  *		value: 		The value being added to the LevelDB database using `put`
  *		opCount: 	The number of operations being performed by `batch`
  *		duration: 	The time taken for the LevelDB query to respond in ms
- *		
+ *
  *		Note: key, value and opCount are undefined for methods other than those *		stated
  */
- 
- 
+
 LeveldownProbe.prototype.metricsEnd = function(probeData, method, methodArgs) {
-    if(probeData && probeData.timer) {
-	    probeData.timer.stop();
-	    if (method == 'put'){
-		    am.emit('leveldown', {time: probeData.timer.startTimeMillis, method: method, key: methodArgs[0], value: methodArgs[1], duration: probeData.timer.timeDelta});
-	    }
-	    else if (method == 'del' || method == 'get'){
-		    am.emit('leveldown', {time: probeData.timer.startTimeMillis, method: method, key: methodArgs[0], duration: probeData.timer.timeDelta});
-	    }
-	    else if(method == 'batch'){
-		    am.emit('leveldown', {time: probeData.timer.startTimeMillis, method: method, opCount: methodArgs[0].length, duration: probeData.timer.timeDelta});
-	    }
+  if (probeData && probeData.timer) {
+    probeData.timer.stop();
+    if (method == 'put') {
+      am.emit('leveldown', {
+        time: probeData.timer.startTimeMillis,
+        method: method,
+        key: methodArgs[0],
+        value: methodArgs[1],
+        duration: probeData.timer.timeDelta,
+      });
+    } else if (method == 'del' || method == 'get') {
+      am.emit('leveldown', {
+        time: probeData.timer.startTimeMillis,
+        method: method,
+        key: methodArgs[0],
+        duration: probeData.timer.timeDelta,
+      });
+    } else if (method == 'batch') {
+      am.emit('leveldown', {
+        time: probeData.timer.startTimeMillis,
+        method: method,
+        opCount: methodArgs[0].length,
+        duration: probeData.timer.timeDelta,
+      });
     }
+  }
 };
 
 /*
  * Heavyweight request probes for leveldown queries
  */
-LeveldownProbe.prototype.requestStart = function (probeData, dbTarget, method, methodArgs) {
-	 req = request.startRequest( 'leveldown', "query" );
-	 req.setContext({leveldown: methodArgs[0]});
+LeveldownProbe.prototype.requestStart = function(
+  probeData,
+  dbTarget,
+  method,
+  methodArgs
+) {
+  // FIXME(sam) req is used as a global to communicate with requestEnd, almost
+  // certainly a bug, what happens if two requests are started before the first
+  // ends?
+  req = request.startRequest('leveldown', 'query');
```

Comment:
I disabled eslint for these lines, but I'm pretty sure from the lines a bit futher down:

```
if (probeData && probeData.req) req.stop(...
```

That just putting `probeData.` before all the `req` would make this work, but since I don't have the setup to test this change, I'll leave that bug fix to you @sjanuary